lib/pry/command_state.rb

```diff
@@ -0,0 +1,29 @@
+require 'ostruct'
+
+class Pry
+  # CommandState is a data structure to hold per-command state.
+  #
+  # Pry commands can store arbitrary state here. This state persists between
+  # subsequent command invocations. All state saved here is unique to the
+  # command.
+  #
+  # @since ?.?.?
+  # @api private
+  class CommandState
+    def self.default
+      @default ||= new
+    end
+
+    def initialize
+      @command_state = {}
+    end
+
+    def state_for(command_name)
+      @command_state[command_name] ||= OpenStruct.new
```

Comment:
Just to let you know, this is one of the reasons Pry::Config::Behavior existed, i see you re going a different way but the original idea would have been to do something like this ..

```ruby
class Pry::CommandState
  include Pry::Config::Behaviour

  def self.state:for(command_name)
    @command_state{command_name] ||= from_hash
  end 

  def self.reset(command_name)
    @command_state = from_hash
  end
end
```

Roughly.

Reply:
Edited original answer.